source/rdt/rdt.go

```diff
@@ -33,52 +30,43 @@ func (s Source) Name() string { return "rdt" }
 func (s Source) Discover() (source.Features, error) {
 	features := source.Features{}
 
-	cmd := exec.Command("bash", "-c", "mon-discovery")
-	if err := cmd.Run(); err != nil {
-		glog.Errorf("support for RDT monitoring was not detected: %v", err)
-	} else {
-		// RDT monitoring detected.
-		features["RDTMON"] = true
-	}
-
-	cmd = exec.Command("bash", "-c", "mon-cmt-discovery")
-	if err := cmd.Run(); err != nil {
-		glog.Errorf("support for RDT CMT monitoring was not detected: %v", err)
-	} else {
-		// RDT CMT monitoring detected.
-		features["RDTCMT"] = true
-	}
-
-	cmd = exec.Command("bash", "-c", "mon-mbm-discovery")
-	if err := cmd.Run(); err != nil {
-		glog.Errorf("support for RDT MBM monitoring was not detected: %v", err)
-	} else {
-		// RDT MBM monitoring detected.
-		features["RDTMBM"] = true
-	}
-
-	cmd = exec.Command("bash", "-c", "l3-alloc-discovery")
-	if err := cmd.Run(); err != nil {
-		glog.Errorf("support for RDT L3 allocation was not detected: %v", err)
-	} else {
-		// RDT L3 cache allocation detected.
-		features["RDTL3CA"] = true
-	}
-
-	cmd = exec.Command("bash", "-c", "l2-alloc-discovery")
-	if err := cmd.Run(); err != nil {
-		glog.Errorf("support for RDT L2 allocation was not detected: %v", err)
-	} else {
-		// RDT L2 cache allocation detected.
-		features["RDTL2CA"] = true
+	// Read cpuid information
+	leaf07h := Cpuid(0x7, 0)
+	leaf0fh := Cpuid(0xf, 0)
+	leaf10h := Cpuid(0x10, 0)
+	leaf07h_1 := Cpuid(0xf, 0x1)
```

Comment:
I would rename variables to something more readable.
Constants like 0x7, 0xf and so on, maybe would be better to declare instead of hardcoding?

Reply:
Yes, these are ugly, but, I didn't change this as I couldn't come up with a better way to rename them. Leaf 07h et al. are directly from the CPU documentation which does not provide much better naming, just *leaf 07h* or *sub-leaf 0*. For the *main* leafs there are some descriptions like "structured extended feature flags enumeration leaf" but I'm not sure how to make something nice out of that

Reply:
I changed the names and used consts in the latest version of this PR